tensorflow_datasets/core/dataset_utils.py

```diff
@@ -177,8 +193,15 @@ def _graph_dataset_iterator(ds_iter, graph=None):
         break
 
 
+@typing.overload
+def as_numpy(dataset: Nested[tf.Tensor],
+             graph: Optional[tf.Graph] = None) -> Nested[np.array]:
+  ...
+
 @api_utils.disallow_positional_args(allowed=["dataset"])
-def as_numpy(dataset, graph=None):
+def as_numpy(dataset: Nested[tf.data.Dataset],
```

Comment:
Have a look at https://docs.python.org/3/library/typing.html#typing.overload for `typing.overload` usage.

The function implementation should not have any anotation.

Reply:
Ohh sorry. I knew that. But did wrong. changed. thanks for suggestion.